### DISCUSSION TITLE
How to set up custom mappings for the `G branch` buffer?

### DISCUSSION THREAD

**Discussion by iamFIREcracker** [tpope/vim-fugitive#2253] [Q&A] [answered]
I would like to set up a new mapping to delete a branch (delete or populate the command prompt accordingly).

Right now, what I do is:

- Press `.`
- Type in `! branch -D`

I see the filetype of the buffer is "git", but I was wondering if there was a way to create a mapping just for _that_ buffer only, and not all the "git" ones.

Thanks.

**Comment by tpope** [comment, marked as answer]
`autocmd User FugitivePager nmap <buffer> ...`

**Reply by iamFIREcracker** [reply]
Great.  And for posterity:

```
augroup fugitive_pager_mappings
    autocmd!

    " <C-F> works better than <C-A> as it would always catch the branch
    " name, even if the cursor is over current branch markers (*/+)
    autocmd User FugitivePager nnoremap <buffer> X :G branch -D <C-R><C-F>
augroup END
```

**Reply by iamFIREcracker** [reply]
Actually, unless I did something stupid by adding that `augroup` to the wrong place (i.e., after/ftplugin/fugitive.vim), it seems to conflict with `:help fugitive_X`.  Does `:G` also triggers the `FugitivePager` event?

**Reply by tpope** [reply]
`:G` with no arguments triggers the `FugitiveIndex` event.
